src/patterns.rs

```diff
@@ -259,56 +259,52 @@ fn rewrite_tuple_pat(pats: &[ptr::P<ast::Pat>],
         pat_vec.insert(pos, dotdot);
     }
 
-    if pat_vec.is_empty() {
-        path_str
```

Comment:
While this is (I think) correct, it also seems inefficient. We could still keep the early return here, we just need to check whether we need `()` or not. However, I confess that I don't see how the code is working for the unit variant case - it looks to me like `TestEnum::Arm2` should get converted to `TestEnum::Arm2()`